zerver/data_import/mattermost.py

```diff
@@ -0,0 +1,540 @@
+        realm_id=realm_id,
+        short_name=short_name,
+        timezone=timezone,
+    )
+
+def convert_user_data(user_handler: UserHandler,
+                      user_id_mapper: IdMapper,
+                      user_data_map: Dict[str, Dict[str, Any]],
+                      realm_id: int,
+                      team_name: str) -> None:
+
+    user_data_list = []
+    for username in user_data_map:
+        user = user_data_map[username]
+        if check_user_in_team(user, team_name) or user["is_mirror_dummy"]:
+            user_data_list.append(user)
+
+    for raw_item in user_data_list:
+        user = process_user(raw_item, realm_id, team_name, user_handler, user_id_mapper)
+        user_handler.add_user(user)
+
+def convert_channel_data(channel_data: List[ZerverFieldsT],
+                         user_data_map: Dict[str, Dict[str, Any]],
+                         subscriber_handler: SubscriberHandler,
+                         stream_id_mapper: IdMapper,
+                         user_id_mapper: IdMapper,
+                         realm_id: int,
+                         team_name: str) -> List[ZerverFieldsT]:
+    channel_data_list = [
+        d
+        for d in channel_data
+        if d['team'] == team_name
+    ]
+
+    channel_members_map = {}  # type: Dict[str, List[str]]
+    channel_admins_map = {}  # type: Dict[str, List[str]]
+
+    def initialize_stream_membership_dicts() -> None:
+        for channel in channel_data:
+            channel_name = channel["name"]
+            channel_members_map[channel_name] = []
+            channel_admins_map[channel_name] = []
+
+        for username in user_data_map:
+            user_dict = user_data_map[username]
+            teams = user_dict["teams"]
+            for team in teams:
+                if team["name"] != team_name:
+                    continue
+                for channel in team["channels"]:
+                    channel_roles = channel["roles"]
+                    channel_name = channel["name"]
+                    if "channel_admin" in channel_roles:
+                        channel_admins_map[channel_name].append(username)
+                    elif "channel_user" in channel_roles:
+                        channel_members_map[channel_name].append(username)
+
+    def get_invite_only(v: str) -> bool:
+        if v == 'O':
+            return False
+        elif v == 'P':
+            return True
+        else:
+            raise Exception('unexpected value')
+
+    streams = []
+    initialize_stream_membership_dicts()
+
+    for channel_dict in channel_data_list:
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(channel_dict['name'])
+        stream_name = channel_dict["name"]
+        invite_only = get_invite_only(channel_dict['type'])
+
+        stream = build_stream(
+            date_created=now,
+            realm_id=realm_id,
+            name=channel_dict['display_name'] or channel_dict['name'],
```

Comment:
What's the story with there being two possible values here?  Is `display_name` potentially HTML or something, a solution for duplicates, or something else?

Reply:
`name` is just a URL friendly version of `display_name`
```
{"type":"channel","channel":{"team":"iago-realm","name":"town-square","display_name":"Town Square","type":"O","header":"","purpose":""}}
{"type":"channel","channel":{"team":"iago-realm","name":"awesome-stream","display_name":"Awesome Stream","type":"O","header":"","purpose":""}}

```

I added the or just in case `display_name` comes value is empty. It probably will lever be empty. I can remove that.

Reply:
yeah, let's just do `display_name` here.